### ISSUE TITLE
Optimize the loop processing of URL decoding

### ISSUE THREAD

**Issue by shane-zfx** [elastic/elasticsearch#110235]
Hey guys, I saw the following content when I was studying the es source code.

https://github.com/elastic/elasticsearch/blob/4a49c08f3417cae17ababbc4134bf8e49be66e98/server/src/main/java/org/elasticsearch/rest/RestUtils.java#L148-L160

I think this method can be optimized in terms of performance efficiency. e.g.
```
for (int i = 0; i < size; i++) {
    final char c = s.charAt(i);
    if (c == '%' || (plusAsSpace && c == '+')) {
         return true;
    } 
}
```

If it's necessary，I'll submit a PR later

**Comment by rjernst** [comment]
@shane-zfx Please feel free to open a PR.

**Comment by shane-zfx** [comment]
Yes, I have submitted the corresponding PR [#110237](https://github.com/elastic/elasticsearch/pull/110237) , please help me review it. :)

**Comment by thecoop** [comment]
Fixed by #110237
